examples/Controllers/Declutching/optimalTimeCalc.m

```diff
@@ -0,0 +1,73 @@
+% This script identifies the dynamics of the float in the respective wave 
+% conditions and determines the optimal proportional gain value for a 
+% passive controller (for regular waves)
+
+close all; clear all; clc;
+
+% Inputs (from wecSimInputFile)
+simu = simulationClass();
+body(1) = bodyClass('../hydroData/rm3.h5');
+waves.height = 1;
+waves.period = 4;
+
+% Load hydrodynamic data for float from BEM
+floatHydro = readBEMIOH5(body.h5File, 1, body.meanDrift);
+
+% Define wave conditions
+H = waves.height;
+A = H/2;
+T = waves.period;
+omega = (1/T)*(2*pi);
+
+% Define excitation force based on wave conditions
+ampSpect = zeros(length(floatHydro.simulation_parameters.w),1);
+[~,closestIndOmega] = min(abs(omega-floatHydro.simulation_parameters.w));
+ampSpect(closestIndOmega) = A;
+FeRao = squeeze(floatHydro.hydro_coeffs.excitation.re(3,:))'*simu.rho*simu.gravity + squeeze(floatHydro.hydro_coeffs.excitation.im(3,:))'*simu.rho*simu.gravity*1j;
+Fexc = ampSpect.*FeRao;
+
+% Define the intrinsic mechanical impedance for the device
+mass = simu.rho*floatHydro.properties.volume;
+addedMass = squeeze(floatHydro.hydro_coeffs.added_mass.all(3,3,:))*simu.rho;
```

Comment:
This seems hard coded to only be the vertical motion and for generalization would have expected it to be selectable; however, you have mentioned that we need coordinate transformations to get the the impedpence at a different location. The MPC WECCCOMP example does something similar by building the dynamic equation of motion about the rotational point. Because most of these examples are custom, I think putting these as examples in the applications case is better than inside of WEC-Sim Master.

Reply:
Yes, currently it is hard-coded for just heave, but I could work on it being selectable DOF. See [Kelley’s comment](https://github.com/WEC-Sim/WEC-Sim/pull/967#issuecomment-1341154287) - the plan is to move the examples to the applications case after review. The reason it was put in WEC-Sim master to be reviewed is to make it function with the controllerClass and library - do you think these are useful in master or should they also be removed and integrated as part of the applications case?